src/pbf_writer.cpp

```diff
@@ -303,19 +303,23 @@ struct pbf_writer::pimpl {
       info->set_visible(t.visible);
     }
     // set the uid and user information, if the user is public
-    std::map<int64_t, int64_t>::const_iterator itr = m_changeset_user_map.find(t.changeset_id);
-    if (itr != m_changeset_user_map.end()) {
-      user_map_t::const_iterator jtr = m_user_map.find(itr->second);
-      if (jtr != m_user_map.end()) {
-        info->set_uid(jtr->first);
-        info->set_user_sid(str_table(jtr->second));
+    if (!m_clean) {
```

Comment:
If the effect of the `m_clean` flag is to make everyone anonymous, could the change be limited to:

```c++
if (m_clean == false && itr != m_changeset_user_map.end()) {
```

Or, if we wish to ensure that `m_changeset_user_map.find` doesn't get called for `m_clean == true` (probably a small optimisation), then something like:

```c++
std::map<int64_t, int64_t>::const_iterator itr = m_changeset_user_map.end();
// when m_clean is set, treat all users as if they were anonymous.
if (m_clean == false) {
  itr = m_changeset_user_map.find(t.changeset_id);
}
if (itr != m_changeset_user_map.end()) {
```

?